### ISSUE TITLE
Include "Location" heading in failing test output for all failures

### ISSUE THREAD

**Issue by darrenburns** [darrenburns/ward#126]
Sometimes failures don't include the "Location" header, which indicates the module and line number an assertion failed on:

<img width="298" alt="Screenshot 2020-02-04 at 22 26 52" src="https://user-images.githubusercontent.com/5740731/73797634-82a2b300-47a8-11ea-8edb-a0c7518cde55.png">

Right now, this header is only printed for assertions containing comparisons e.g. `assert x == y`, `assert x in y`, etc (because these are the types of assertions that are rewritten using AST transformation). This header should be printed for all AssertionErrors. This doesn't mean we need to rewrite all assertions for now, we could probably just catch the raised AssertionErrors.

**Comment by jayeshathila** [comment]
Hi @darrenburns,
If you understood you right you mean to change the `output_test_failed_location` method to something: 
```       
    def output_test_failed_location(self, test_result: TestResult):
        if isinstance(test_result.error, TestFailure):
            print(
                indent(colored("Location:", color="cyan", attrs=["bold"]), INDENT),
                f"{test_result.test.path.relative_to(Path.cwd())}:{test_result.error.error_line}",
            )
        elif isinstance(test_result.error, AssertionError):
            print(
                indent(colored("Location:", color="cyan", attrs=["bold"]), INDENT),
                f"{test_result.test.path.relative_to(Path.cwd())}:{test_result.error}",
            )
```
(Haven't passed on line number yet in above change though)

https://github.com/darrenburns/ward/blob/master/ward/terminal.py#L501

Let me know if this is what you intended or something else.

**Comment by darrenburns** [comment]
Hey @jayeshathila, yes that looks like what I was thinking :)

**Comment by jayeshathila** [comment]
Cool. Will raise a PR :)

**Comment by darrenburns** [comment]
This is available in Ward 0.39.0b0 and beyond.
